mlrun/api/runtime_handlers/kubejob.py

```diff
@@ -11,15 +11,185 @@
+                if code:
+                    raise ValueError("cannot specify both code and source archive")
+                args += ["--source", runtime.spec.build.source]
+                if runtime.spec.workdir:
+                    # set the absolute/relative path to the cloned code
+                    args += ["--workdir", runtime.spec.workdir]
+
+            if command:
+                args += [command]
+
+            if runtime.spec.args:
+                if not command:
+                    # * is a placeholder for the url argument in the run CLI command,
+                    # where the code is passed in the `MLRUN_EXEC_CODE` meaning there is no "actual" file to execute
+                    # until the run command will create that file from the env param.
+                    args += ["*"]
+                args = args + runtime.spec.args
+
+            command = "mlrun"
+        else:
+            command = command.format(**run.spec.parameters)
+            if runtime.spec.args:
+                args = [arg.format(**run.spec.parameters) for arg in runtime.spec.args]
+
+        extra_env = [{"name": k, "value": v} for k, v in extra_env.items()]
+        return command, args, extra_env
+
+    @staticmethod
+    def _resolve_workdir(runtime: mlrun.runtimes.KubejobRuntime):
+        """
+        The workdir is relative to the source root, if the source is not loaded on run then the workdir
+        is relative to the clone target dir (where the source was copied to).
+        Otherwise, if the source is loaded on run, the workdir is resolved on the run as well.
+        If the workdir is absolute, keep it as is.
+        """
+        workdir = runtime.spec.workdir
+        if runtime.spec.build.source and runtime.spec.build.load_source_on_run:
+            # workdir will be set AFTER the clone which is done in the pre-run of local runtime
+            return None
+
+        if workdir and os.path.isabs(workdir):
+            return workdir
+
+        if runtime.spec.clone_target_dir:
+            workdir = workdir or ""
+            workdir = workdir.removeprefix("./")
+
+            return os.path.join(runtime.spec.clone_target_dir, workdir)
+
+        return workdir
+
+    @staticmethod
+    def _generate_runtime_env(
+        runtime: mlrun.runtimes.BaseRuntime, run: mlrun.run.RunObject
+    ) -> typing.Dict:
+        """
+        Prepares all available environment variables for usage on a runtime
+        Data will be extracted from several sources and most of them are not guaranteed to be available
+
+        :param runtime: The ad-hoc runtime object
+        :param run:     Run context object (RunObject) with run metadata and status
+        :return:        Dictionary with all the variables that could be parsed
+        """
+        runtime_env = {
+            "MLRUN_DEFAULT_PROJECT": runtime.metadata.project or config.default_project
+        }
+        if run:
+            runtime_env["MLRUN_EXEC_CONFIG"] = run.to_json()
```

Comment:
@yaelgen alert

Reply:
@yaelgen ended up not moving it because it is needed in the client